scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/StaxJobFactory.java

```diff
@@ -479,6 +480,56 @@ private void createJobVariables(XMLStreamReader cursorVariables)
         return variablesMap;
     }
 
+     /**
+      * Create a map of variables from XML variables.
+      * Leave the method with the cursor at the end of 'ELEMENT_VARIABLES' tag
+      *
+      * @param cursorVariables the streamReader with the cursor on the 'ELEMENT_VARIABLES' tag.
+      * @return the map in which the variables were added.
+      * @throws JobCreationException
+      */
+      private Map<String, TaskVariable> createTaskVariables(XMLStreamReader cursorVariables) 
+              throws JobCreationException {
+         Map<String, TaskVariable> variablesMap = new HashMap<>();
+         try {
+             int eventType;
+             while (cursorVariables.hasNext()) {
+                 eventType = cursorVariables.next();
+                 switch (eventType) {
+                     case XMLEvent.START_ELEMENT:
+                         if (XMLTags.VARIABLE.matches(cursorVariables.getLocalName())) {
+                             Map<String, String> attributesAsMap = getAttributesAsMap(cursorVariables);
+
+                             //TODO checks for data consistency
+                             TaskVariable taskVariable = new TaskVariable();
+                             taskVariable.setName(attributesAsMap.get(XMLAttributes.VARIABLE_NAME.getXMLName()));
+                             taskVariable.setValue(attributesAsMap.get(XMLAttributes.VARIABLE_VALUE.getXMLName()));
+                             taskVariable.setModel(attributesAsMap.get(XMLAttributes.VARIABLE_MODEL.getXMLName()));
+                             taskVariable.setJobInherited(Boolean.valueOf(attributesAsMap.get(XMLAttributes.VARIABLE_JOB_INHERITED.getXMLName())));
+
+                             variablesMap.put(taskVariable.getName(), taskVariable);
+                         }
+                         break;
+                     case XMLEvent.END_ELEMENT:
+                         if (XMLTags.VARIABLES.matches(cursorVariables.getLocalName())) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS134)